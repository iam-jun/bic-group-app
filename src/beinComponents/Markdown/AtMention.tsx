import React from 'react';
import {
  StyleProp, Text, TextStyle, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import isEmpty from 'lodash/isEmpty';

import { IMarkdownAudience } from '~/interfaces/IPost';

import { fontFamilies } from '~/theme/fonts';
import useMentionInputStore from '../inputs/MentionInput/store';
import IMentionInputState from '../inputs/MentionInput/store/Interface';
import { useKeySelector } from '~/hooks/selector';

interface Props {
  mentions?: any[];
  mentionName: string;
  style?: StyleProp<TextStyle>;
  onPress?: (audience: IMarkdownAudience, e: any) => void;
}

const AtMention = ({
  mentionName,mentions, style, onPress,
}: Props) => {
  const theme: ExtendedTheme = useTheme();
  const styles = createStyles(theme);

  let audience = mentions?.[mentionName];
  
  const tempSelectedUser = useMentionInputStore(
    (state: IMentionInputState) => state.tempSelected?.[mentionName]
  )

  if (!audience) {
    audience = tempSelectedUser || {};
  }
  const name = audience?.fullname || audience?.data?.fullname;
  const withAudience = !isEmpty(audience);

  const _onPress = (e: any) => {
    onPress?.(audience, e);
  };

  return (
    <Text
      testID="text_mention"
      style={withAudience ? style : styles.baseStyle}
      onPress={withAudience ? _onPress : undefined}
    >
      {`@${
        name || mentionName
      }`}
    </Text>
  );
};

export default AtMention;

const createStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    baseStyle: {
      color: colors.neutral80,
      fontFamily: fontFamilies.BeVietnamProLight,
    },
  });
};
