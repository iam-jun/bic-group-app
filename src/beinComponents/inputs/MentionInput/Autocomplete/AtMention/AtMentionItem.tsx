import React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import {useTheme} from 'react-native-paper';

import Avatar from '~/beinComponents/Avatar';
import Text from '~/beinComponents/Text';
import images from '~/resources/images';
import {ITheme} from '~/theme/interfaces';
import spacing from '~/theme/spacing';

interface Props {
  testID?: string;
  item: any;
  onPress: (item: any) => void;
}

const AtMentionItem = ({item, onPress}: Props) => {
  const theme = useTheme() as ITheme;
  const styles = createStyles(theme);

  const _onPressItem = () => {
    onPress(item);
  };

  const renderMentionAll = () => {
    return (
      <View>
        <TouchableOpacity
          testID="at_mention_item.item_all"
          onPress={_onPressItem}>
          <View style={styles.mentionAll}>
            <Text.ButtonBase style={styles.textMentionAll}>
              @all
            </Text.ButtonBase>
            <Text.Subtitle useI18n>common:title_mention_all</Text.Subtitle>
          </View>
        </TouchableOpacity>
      </View>
    );
  };

  if (item.username === 'all') return renderMentionAll();

  return (
    <View testID={'at_mention_item'} style={styles.mentionContainer}>
      <TouchableOpacity
        style={styles.container}
        onPress={_onPressItem}
        testID={'at_mention_item.touchable'}>
        <Avatar.Medium
          style={styles.avatar}
          source={item.avatar || item.icon}
          placeholderSource={images.img_user_avatar_default}
        />
        <Text>{item.name || item.fullname}</Text>
      </TouchableOpacity>
    </View>
  );
};

const createStyles = (theme: ITheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      marginHorizontal: spacing.margin.base,
      marginVertical: spacing.margin.small,
    },
    mentionContainer: {
      backgroundColor: colors.background,
    },
    mentionAll: {
      flexDirection: 'row',
      padding: spacing?.padding.base,
      alignItems: 'center',
    },
    textMentionAll: {
      marginEnd: spacing?.margin.base,
    },
  });
};

export default AtMentionItem;
