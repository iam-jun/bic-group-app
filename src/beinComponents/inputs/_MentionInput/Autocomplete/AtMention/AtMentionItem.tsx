import React from 'react';
import {StyleSheet, TouchableOpacity} from 'react-native';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import Avatar from '~/beinComponents/Avatar';
import Div from '~/beinComponents/Div';
import Text from '~/beinComponents/Text';
import {AT_MENTION_REGEX} from '~/constants/autocomplete';
import {useKeySelector} from '~/hooks/selector';
import images from '~/resources/images';
import {ITheme} from '~/theme/interfaces';
import actions from '../../redux/actions';

const AtMentionItem = ({item}: any) => {
  const dispatch = useDispatch();
  const {text, cursorPosition, highlightItem} = useKeySelector('mentionInput');

  const theme = useTheme() as ITheme;
  const {colors} = theme;
  const styles = createStyles(theme);

  const backgroundColor =
    highlightItem?.id && item?.id === highlightItem?.id
      ? colors.placeholder
      : colors.background;

  const onHoverItem = () => {
    dispatch(actions.sethHighlightItem(item));
  };

  const onLeaveItem = () => {
    dispatch(actions.sethHighlightItem(null));
  };

  const completeMention = (mention: string) => {
    //    const {cursorPosition, isSearch, onChangeText, value} = this.props;
    const mentionPart = text.substring(0, cursorPosition);

    let completedDraft = mentionPart.replace(AT_MENTION_REGEX, `@${mention} `);

    if (text.length > cursorPosition) {
      completedDraft += text.substring(cursorPosition);
    }

    dispatch(actions.setText(completedDraft));
    dispatch(actions.setData([]));
  };

  const _onPressItem = () => {
    completeMention(item.username);
  };

  return (
    <Div
      style={{backgroundColor}}
      onMouseOver={onHoverItem}
      onMouseLeave={onLeaveItem}>
      <TouchableOpacity style={styles.container} onPress={_onPressItem}>
        <Avatar.Medium
          style={styles.avatar}
          source={item.avatar || item.icon}
          placeholderSource={images.img_user_avatar_default}
        />
        <Text>{item.name || item.fullname}</Text>
      </TouchableOpacity>
    </Div>
  );
};

const createStyles = (theme: ITheme) => {
  const {spacing} = theme;
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      marginHorizontal: spacing.margin.base,
      marginVertical: spacing.margin.small,
    },
  });
};

export default AtMentionItem;
