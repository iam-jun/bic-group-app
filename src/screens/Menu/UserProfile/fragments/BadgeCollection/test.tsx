import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
