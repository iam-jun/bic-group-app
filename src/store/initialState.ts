import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
