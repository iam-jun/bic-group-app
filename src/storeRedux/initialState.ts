import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
