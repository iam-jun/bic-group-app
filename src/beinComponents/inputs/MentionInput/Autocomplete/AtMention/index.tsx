import {isEmpty} from 'lodash';
import React, {useEffect, useRef} from 'react';
import {
  ActivityIndicator,
  DeviceEventEmitter,
  FlatList,
  StyleSheet,
  View,
} from 'react-native';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import Text from '~/beinComponents/Text';
import {useKeySelector} from '~/hooks/selector';
import {ITheme} from '~/theme/interfaces';
import {AutocompleteProps} from '..';
import {
  checkRunSearch,
  completeMention,
  ICursorPositionChange,
} from '../../helper';
import actions from '../../redux/actions';
import AtMentionItem from './AtMentionItem';

const DEFAULT_INDEX = -1;

type Props = Partial<AutocompleteProps>;

const AtMention = ({
  showSpectialItems,
  emptyContent,
  cursorPosition,
}: Props) => {
  const dispatch = useDispatch();

  const {data, loading, highlightIndex} = useKeySelector('mentionInput');

  const text = useRef('');

  const listRef = useRef<any>();

  const theme = useTheme() as ITheme;
  const {colors} = theme;

  const styles = createStyles(theme);

  useEffect(() => {
    const listener = DeviceEventEmitter.addListener(
      'autocomplete-on-selection-change',
      onCursorPositionChange,
    );
    return () => {
      listener?.remove?.();
    };
  }, []);

  useEffect(() => {
    const listener = DeviceEventEmitter.addListener(
      'autocomplete-on-key-press',
      handleMentionKey,
    );
    return () => {
      listener?.remove?.();
    };
  }, [data, highlightIndex]);

  const onCursorPositionChange = ({
    position,
    value,
    groupIds,
  }: ICursorPositionChange) => {
    text.current = value;
    checkRunSearch(value.substring(0, position), groupIds, dispatch);
  };

  const _completeMention = (item: any) => {
    completeMention({item, dispatch, text: text.current, cursorPosition});
  };

  const handleMentionKey = (event: any) => {
    if (!isEmpty(data)) {
      event.preventDefault();
      const {key} = event || {};
      if (key === 'Enter' && data?.[highlightIndex]) {
        _completeMention(data[highlightIndex]);
        return;
      }
      const step = key === 'ArrowUp' ? -1 : 1;
      const min = 0;
      let newIndex =
        highlightIndex === DEFAULT_INDEX ? min : highlightIndex + step;
      if (newIndex >= data.length) {
        newIndex = min;
      }
      if (newIndex < min) {
        newIndex = data.length - 1;
      }

      const newHighlightItem: any = data?.[newIndex];

      if (newIndex >= 0 && newIndex < data?.length) {
        listRef.current?.scrollToIndex({
          index: newIndex,
          viewPosition: 0.5,
        });
      }
      dispatch(actions.sethHighlightIndex(newIndex));
      dispatch(actions.sethHighlightItem(newHighlightItem));
    }
  };

  const renderEmpty = () => {
    return (
      <View style={styles.emptyContainer}>
        {loading ? (
          <ActivityIndicator
            testID="at_mention.loading"
            color={colors.disabled}
          />
        ) : (
          <Text.H6 testID="at_mention.empty_content" style={styles.textEmpty}>
            {emptyContent}
          </Text.H6>
        )}
      </View>
    );
  };

  const renderItem = ({item, index}: {item: any; index: number}) => {
    return (
      <AtMentionItem
        testID={`at_mention.item_${index}`}
        item={item}
        onPress={_completeMention}
      />
    );
  };

  const _data = showSpectialItems ? [{username: 'all'}, ...data] : data;

  return (
    <FlatList
      testID="at_mention"
      ref={listRef}
      data={_data}
      keyExtractor={item => `list-mention-${item.username}`}
      renderItem={renderItem}
      ListEmptyComponent={renderEmpty}
    />
  );
};

const createStyles = (theme: ITheme) => {
  const {colors, spacing} = theme;

  return StyleSheet.create({
    textEmpty: {
      color: colors.textDisabled,
      padding: spacing.padding.tiny,
      margin: spacing.margin.small,
    },
    emptyContainer: {
      minHeight: 40,
      justifyContent: 'center',
    },
  });
};

export default AtMention;
``;
