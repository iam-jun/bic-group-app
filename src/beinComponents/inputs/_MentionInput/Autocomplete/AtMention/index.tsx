import {isEmpty} from 'lodash';
import React, {useEffect, useRef} from 'react';
import {
  ActivityIndicator,
  DeviceEventEmitter,
  FlatList,
  Platform,
  StyleSheet,
  View,
} from 'react-native';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import Text from '~/beinComponents/Text';
import {AT_MENTION_REGEX} from '~/constants/autocomplete';
import {useKeySelector} from '~/hooks/selector';
import {ITheme} from '~/theme/interfaces';
import {AutocompleteProps} from '../../Autocomplete';
import {getMatchTermForAtMention} from '../../helper';
import actions from '../../redux/actions';
import AtMentionItem from './AtMentionItem';

const DEFAULT_INDEX = -1;

interface Props extends Partial<AutocompleteProps> {}

const AtMention = ({
  showSpectialItems,
  emptyContent,
  onCompletePress,
}: Props) => {
  const dispatch = useDispatch();

  const {data, isLoading, highlightIndex} = useKeySelector('mentionInput');

  const text = useRef('');
  const cursorPosition = useRef(0);

  const listRef = useRef<any>();

  const theme = useTheme() as ITheme;
  const {colors} = theme;

  const styles = createStyles(theme);

  useEffect(() => {
    const listener = DeviceEventEmitter.addListener(
      'autocomplete-on-selection-change',
      onCursorPositionChange,
    );
    return () => {
      listener?.remove?.();
    };
  }, []);

  useEffect(() => {
    const listener = DeviceEventEmitter.addListener(
      'autocomplete-on-key-press',
      handleMentionKey,
    );
    return () => {
      listener?.remove?.();
    };
  }, [data, highlightIndex]);

  const onCursorPositionChange = ({
    position,
    value,
    groupIds,
  }: {
    position: number;
    value: string;
    groupIds: string;
  }) => {
    text.current = value;
    cursorPosition.current = position;

    const _text = value.substring(0, position);
    const _matchTerm = getMatchTermForAtMention(_text, false);
    if (_matchTerm !== null && !_matchTerm.endsWith(' ')) {
      dispatch(actions.runSearch({group_ids: groupIds, key: _matchTerm}));
    } else {
      dispatch(actions.setData([]));
    }
  };

  const completeMention = (item: any) => {
    const mention = item.username;
    const mentionPart = text.current.substring(0, cursorPosition.current);

    let completedDraft = mentionPart.replace(AT_MENTION_REGEX, `@${mention} `);

    if (text.current.length > cursorPosition.current) {
      completedDraft += text.current.substring(cursorPosition.current);
    }
    onCompletePress?.(completedDraft);
    dispatch(actions.setData([]));
  };

  const handleMentionKey = (event: any) => {
    if (!isEmpty(data)) {
      event.preventDefault();
      const {key} = event || {};
      if (key === 'Enter' && data?.[highlightIndex]) {
        completeMention(data?.[highlightIndex]);
        return;
      }
      const step = key === 'ArrowUp' ? -1 : 1;
      const min = 0;
      let newIndex =
        highlightIndex === DEFAULT_INDEX ? min : highlightIndex + step;
      if (newIndex >= data.length) {
        newIndex = min;
      }
      if (newIndex < min) {
        newIndex = data.length - 1;
      }

      const newHighlightItem: any = data?.[newIndex];

      if (newIndex >= 0 && newIndex < data?.length) {
        listRef.current?.scrollToIndex({
          index: newIndex,
          viewPosition: 0.5,
        });
      }
      dispatch(actions.sethHighlightIndex(newIndex));
      dispatch(actions.sethHighlightItem(newHighlightItem));
    }
  };

  const renderEmpty = () => {
    return (
      <View style={styles.emptyContainer}>
        {isLoading ? (
          <ActivityIndicator color={colors.disabled} />
        ) : (
          <Text.H6 style={styles.textEmpty}>{emptyContent}</Text.H6>
        )}
      </View>
    );
  };

  const renderItem = ({item}: {item: any; index: number}) => {
    return <AtMentionItem item={item} onPress={completeMention} />;
  };

  if (isEmpty(data)) return null;

  const _data = showSpectialItems ? [{username: 'all'}, ...data] : data;

  return (
    <FlatList
      ref={listRef}
      data={_data}
      keyExtractor={item => `list-mention-${item.id}`}
      renderItem={renderItem}
      ListEmptyComponent={renderEmpty}
    />
  );
};

const createStyles = (theme: ITheme) => {
  const {colors, spacing} = theme;

  return StyleSheet.create({
    textTitle: {
      marginVertical: spacing.margin.small,
      marginHorizontal: spacing.margin.base,
      color: colors.textSecondary,
    },
    textEmpty: {
      color: colors.textDisabled,
      padding: spacing.padding.tiny,
      margin: spacing.margin.small,
    },
    emptyContainer: {
      minHeight: 40,
      justifyContent: 'center',
    },
    hidden: {
      height: 0,
      flex: undefined,
      marginTop: 0,
      marginBottom: 0,
      paddingTop: 0,
      paddingBottom: 0,
      borderWidth: 0,
      ...Platform.select({
        web: {
          border: 'none',
          marginTop: '0px important',
          marginBottom: '0px important',
        },
      }),
    },
  });
};

export default AtMention;
