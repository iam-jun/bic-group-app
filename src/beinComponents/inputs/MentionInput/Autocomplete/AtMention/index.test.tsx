import React from 'react';
import {cleanup} from '@testing-library/react-native';

import {configureStore, renderWithRedux} from '~/test/testUtils';
import initialState from '~/store/initialState';
import AtMention from '.';
import {colors} from '~/theme';
import {StyleSheet} from 'react-native';

afterEach(cleanup);

describe('AtMention component', () => {
  const baseProps = {
    showSpectialItems: false,
    emptyContent: 'emptyContent',
    cursorPosition: 0,
    onCompletePress: jest.fn(),
  };

  const mockStore = configureStore([]);

  it(`renders correctly`, async () => {
    const store = mockStore(initialState);
    const wrapper = renderWithRedux(<AtMention {...baseProps} />, store);

    const rendered = wrapper.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it(`should hide "AtMention" with empty content when data is empty`, async () => {
    const storeData = {...initialState};
    storeData.mentionInput.data = [];
    const store = mockStore(storeData);
    const wrapper = renderWithRedux(<AtMention {...baseProps} />, store);
    const component = wrapper.getByTestId('at_mention.empty_content');

    expect(component).not.toBeNull();

    const flattenedStyle = StyleSheet.flatten(component.props.style);
    expect(flattenedStyle.color).toBe(colors.light.colors.textDisabled);
  });

  it(`should show "AtMention" with loading`, async () => {
    const storeData = {...initialState};
    storeData.mentionInput.data = [] as any;
    storeData.mentionInput.loading = true;
    const store = mockStore(storeData);
    const wrapper = renderWithRedux(<AtMention {...baseProps} />, store);
    const component = wrapper.getByTestId('at_mention.loading');

    expect(component).not.toBeNull();

    expect(component.props.color).toBe(colors.light.colors.disabled);
  });

  it(`should hide loading`, async () => {
    const storeData = {...initialState};
    storeData.mentionInput.data = [] as any;
    storeData.mentionInput.loading = false;
    const store = mockStore(storeData);
    const wrapper = renderWithRedux(<AtMention {...baseProps} />, store);
    const component = wrapper.queryByTestId('at_mention.loading');

    expect(component).toBeNull();
  });

  it(`should show "AtMention" with item all`, async () => {
    const storeData = {...initialState};
    const store = mockStore(storeData);
    const props = {...baseProps, showSpectialItems: true};
    const wrapper = renderWithRedux(<AtMention {...props} />, store);
    const component = wrapper.getByTestId('at_mention');

    expect(component.props.data[0].username).toBe('all');
  });

  it(`should show "AtMention" with items`, async () => {
    const storeData = {...initialState};
    const item = {username: 'test', name: 'test'};
    storeData.mentionInput.data = [item] as any;
    const store = mockStore(storeData);
    const wrapper = renderWithRedux(<AtMention {...baseProps} />, store);
    const component = wrapper.getByTestId('at_mention_item');

    expect(component).not.toBeNull();
  });
});
