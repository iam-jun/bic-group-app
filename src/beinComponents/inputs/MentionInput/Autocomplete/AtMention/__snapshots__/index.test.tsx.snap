// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtMention component renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  testID="at_mention"
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "minHeight": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#181C32",
              "fontFamily": "BeVietnamPro-SemiBold",
              "fontSize": 13,
            },
            Object {},
            Object {
              "color": "#A4A9B7",
              "margin": 8,
              "padding": 4,
            },
          ]
        }
        testID="at_mention.empty_content"
      >
        emptyContent
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
